src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.MinMagnitude.cs

```diff
@@ -85,148 +85,72 @@ public static void MinMagnitude<T>(ReadOnlySpan<T> x, T y, Span<T> destination)
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
             public static Vector128<T> Invoke(Vector128<T> x, Vector128<T> y)
             {
-                Vector128<T> xMag = Vector128.Abs(x), yMag = Vector128.Abs(y);
-
-                Vector128<T> result =
-                    Vector128.ConditionalSelect(Vector128.Equals(yMag, xMag),
-                        Vector128.ConditionalSelect(IsNegative(y), y, x),
-                        Vector128.ConditionalSelect(Vector128.LessThan(yMag, xMag), y, x));
+#if NET9_0_OR_GREATER
+                return Vector128.MinMagnitude(x, y);
+#else
 
-                if (typeof(T) == typeof(sbyte) || typeof(T) == typeof(short) || typeof(T) == typeof(int) || typeof(T) == typeof(long) || typeof(T) == typeof(nint))
+                if ((typeof(T) == typeof(float)) || (typeof(T) == typeof(double)))
```

Comment:
Why is this restricting the number of supported types?

Reply:
It's not restricting the number of supported types, it's applying specialization where a difference in behavior is required and deferring down to an existing correct implementation otherwise.

`MinMagnitude` exists to propagate NaN, `MinMagnitudeNumber` exists to not propagate `NaN`. Thus `MinMagnitude` only needs to exist for supported `T` that have some `NaN` (i.e. `float` and `double`) and otherwise can defer to `MinMagnitudeNumber`